### PR TITLE
Rendering

### PR DESCRIPTION
**Pull request by whatisaphone**
Add rendering

### CODE REVIEW

File: examples/rendering.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            &green,
+        );
+        group.draw_line_2d(
+            center_x,
+            center_y,
+            center_x + (ms as f32 * 2.0 * PI / 1000.0 - PI / 2.0).cos() * r_ms,
+            center_y + (ms as f32 * 2.0 * PI / 1000.0 - PI / 2.0).sin() * r_ms,
+            &green,
+        );
+        group.render()?;
```
Comment (ehsanul) [inline examples/rendering.rs]: Nice, you seem to have made this pretty ergonomic.

File: src/render.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+/// #
+/// # fn main() -> Result<(), Box<Error>> {
+/// let rlbot = rlbot::init()?;
+/// let group = rlbot.begin_render_group(1234);
+/// group.render()?;
+/// # Ok(())
+/// # }
+/// ```
+pub struct RenderGroup<'a> {
+    rlbot: &'a RLBot,
```
Comment (ehsanul) [inline src/render.rs]: I guess a consequence of this is that the user has to make sure that the `RenderGroup` goes out of scope before they can use `rlbot` again, right? And you can't have two active `RenderGroup`s at the same time.

Not the end of the world, but given that only the `render` call needs this, might make sense to make `render` just take a `&RLBot`, so the borrow is limited to the minimum scope.

File: examples/rendering.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    // Wait for the match to start. `packets.next()` sleeps until the next
+    // packet is available, so this loop will not roast your CPU :)
+    while !packets.next()?.GameInfo.RoundActive {}
+
+    loop {
+        let packet = packets.next()?;
+        let mut total_ms = (packet.GameInfo.TimeSeconds * 1000.0) as i32;
+        let ms = total_ms % 1000;
+        total_ms -= ms;
+        let sec = total_ms / 1000 % 60;
```
Comment (ehsanul) [inline examples/rendering.rs]: It's probably just me, but I always forget precedence rules for stuff like this and opt to add parens for extra clarity.

File: src/render.rs [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    /// A color can only be used with the `RenderGroup` that created it.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use rlbot::RenderGroup;
+    /// # let mut group: RenderGroup = unsafe { ::std::mem::uninitialized() };
+    /// let transbluecent = group.color_argb(127, 0, 0, 255);
+    /// ```
+    pub fn color_argb(&mut self, a: u8, r: u8, g: u8, b: u8) -> Color<'a> {
```
Comment (ehsanul) [inline src/render.rs]: This order of arguments matches the order it's defined in, but personally I find rgba more intuitive. This is subjective though, so feel free to ignore this.

Comment (whatisaphone) [reply src/render.rs]: Thanks for reviewing this. Like you said this is subjective, and in cases like this where there's not a clear reason for changing I think it's better to stick with the defined order, so I think I'll keep it as is.

**Pull request merged**